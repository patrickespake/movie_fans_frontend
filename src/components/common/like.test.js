import React from "react";
import { shallow } from "enzyme";
import Like from "./Like";

describe("Like component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Like />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders an i element", () => {
    expect(wrapper.find("i").length).toBe(1);
  });

  it("has the correct class when liked prop is true", () => {
    wrapper.setProps({ liked: true });
    expect(wrapper.find("i").hasClass("fa-heart")).toBe(true);
  });

  it("has the correct class when liked prop is false", () => {
    wrapper.setProps({ liked: false });
    expect(wrapper.find("i").hasClass("fa-heart-o")).toBe(true);
  });

  it("calls the onClick prop when clicked", () => {
    const onClick = jest.fn();
    wrapper.setProps({ onClick });
    wrapper.find("i").simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
});
