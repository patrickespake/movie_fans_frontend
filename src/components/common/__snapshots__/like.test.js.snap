// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like component renders correctly 1`] = `ShallowWrapper {}`;
